### ISSUE TITLE
Build failure in channelmixerrgb.c

### ISSUE THREAD

**Issue by 1div0** [darktable-org/darktable#8339]
**Compilation with GNU C/C++ 11.0.0 failed**

```
In file included from /1TB/usr/src/github.com/1div0/darktable/build/lib64/darktable/plugins/introspection_channelmixerrgb.c:164:
/1TB/usr/src/github.com/1div0/darktable/src/iop/channelmixerrgb.c: In function ‘_extract_patches’:
/1TB/usr/src/github.com/1div0/darktable/src/iop/channelmixerrgb.c:1337:5: error: ‘dot_product’ accessing 16 bytes in a region of size 12 [-Werror=stringop-overflow=]
 1337 |     dot_product(patches + k * 4, RGB_to_XYZ, XYZ);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1: all warnings being treated as errors
gmake[2]: *** [lib64/darktable/plugins/CMakeFiles/channelmixerrgb.dir/build.make:88: lib64/darktable/plugins/CMakeFiles/channelmixerrgb.dir/introspection_channelmixerrgb.c.o] Error 1
```

```
€ git clone --recursive git@github.com:1div0/darktable.git
€ cd darktable
€ ./build.sh --prefix /opt/darktable+bmff --build-type Release --install --sudo
```
Fedora 33 (Workstation Edition)

However compiler is from upcoming F34

**Comment by kmilos** [comment]
@TurboGit looks like also addressed by 7825e11076d59b2c96c3e3adfd12100eaf7b8401 and c6c0f568b7bbcc9a761e3f37d96c5907e863fc65

**Comment by TurboGit** [comment]
Probably yes, closing.
